pkg/kapp/resources/mod_path.go

```diff
@@ -20,15 +20,16 @@ type ResourceModWithMultiple interface {
 type Path []*PathPart
 
 type PathPart struct {
-	MapKey     *string
-	ArrayIndex *PathPartArrayIndex
+	MapKey        *string
+	IndexAndRegex *PathPartIndexAndRegex
```

Comment:
I feel that Regex could be a different value here. 
- It would reduce a lot of noise in the PR.
- Index and Regex might have similar use cases, but they are actually different in the sense that one works for arrays while the other doesn't, so it does make sense to have a different structure for Regex.

Reply:
If everyone is aligned to have it as a different value I can make that change but this will be a significant change again and have to look into parsing the regex part separately.